content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+
+When you introspect your database, Prisma will take the database type of each table column and represent it in your Prisma schema using the correct Prisma type for the corresponding model field. If the database type is not the default database type for that Prisma scalar type, Prisma will also add a native type attribute.
+
+As an example, take a `User` table in a PostgreSQL database, with:
+
+- an `id` column with a data type of `serial`
+- a `name` column with a data type of `text`
+- an `isActive` column with a data type of `boolean`
+
+You can create this with the following SQL command:
+
+```sql
+CREATE TABLE "public"."User" (
+  id serial PRIMARY KEY NOT NULL,
+  name text NOT NULL,
+  "isActive" boolean NOT NULL
+);
+```
+
+Introspect your database with the following command run from the root directory of your project:
+
+```terminal
+npx prisma db pull
+```
+
+You will get the following Prisma schema:
+
+```prisma file=schema.prisma
+model User {
+  id       Int     @id @default(autoincrement())
+  name     String
+  isActive Boolean
+}
+```
+
+The `id`, `name` and `isActive` columns in the database are mapped respectively to the `Int`, `String` and `Boolean` Prisma types respectively. These are the default database types for these Prisma types, so Prisma does not add any native type attributes.
+
+Now add a `createdAt` column to your database with a data type of `date` by running the following SQL command:
+
+```sql
+ALTER TABLE "public"."User"
+ADD COLUMN "createdAt" date NOT NULL;
+```
+
+Now introspect your database again:
+
+```terminal
+npx prisma db pull
+```
+
+Your Prisma schema now includes the new `createdAt` field with a Prisma type of `DateTime`. The `createdAt` field also has a `db.Date` native type attribute, because `date` is not the default type for the `DateTime` type:
+
+```prisma file=schema.prisma highlight=5;add
+model User {
+  id        Int      @id @default(autoincrement())
+  name      String
+  isActive  Boolean
+  createdAt DateTime @db.Date
+}
+```
+
+## How to use types when you apply schema changes to your database
+
+When you apply schema changes to your database using `migrate dev` or `db push`, Prisma will use both the Prisma scalar type of each field and any native attribute it has to determine the correct database type for the corresponding column in the database.
+
+As an example, create a Prisma schema with the following `Post` model:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime
+  updatedAt DateTime @db.Date
+}
+```
+
+This `Post` model has:
+
+- an `id` field with a Prisma type of `Int`
+- a `title` field with a Prisma type of `String`
+- a `createdAt` field with a Prisma type of `DateTime`
+- an `updatedAt` field with a Prisma type of `DateTime` and a `@db.Date` native type attribute
+
+Now apply these changes to an empty PostgreSQL database with the following command, run from the root directory of your project:
```

Comment:
To run this the schema is missing the `datasource` bit. Might not be relevant.

Reply:
I think this is now ok with the datasource added and explained at the top